test/unit/tools/unified_spec_runner.test.ts

```diff
@@ -37,7 +37,7 @@ describe('Unified Spec Runner', function () {
           it('throws AssertionError when it finds extra keys', function () {
             actual =
               '{"data": {"$numberLong": "100"}, "a": {"$numberInt": "10"}, "b": {"$numberInt": "100"}}';
-            expect(() => resultCheckSpy(actual, expected, entitiesMap, [])).to.throw(
+            expect(() => resultCheckSpy(actual, expected, entitiesMap, [], true)).to.throw(
```

Comment:
I'm surprised this test was previously passing since `resultCheck`'s `checkExtraKeys` variable defaulted to `false`. Can we add a test asserting the inverse of this one?

Reply:
I've added another test here for the inverse.